experimental/trusted_shuffler/trusted_shuffler/src/lib.rs

```diff
@@ -16,4 +16,131 @@
 
 //! Trusted Shuffler implementation.
 
-// TODO(#2614): Implement Trusted Shuffler logic.
+#[cfg(test)]
+mod tests;
+
+use anyhow::{anyhow, Context};
+use futures::future::join_all;
+use rand::{seq::SliceRandom, thread_rng};
+use std::{
+    future::Future,
+    marker::{Send, Sync},
+    mem::replace,
+    ops::DerefMut,
+    sync::{Arc, Mutex},
+};
+use tokio::sync::oneshot;
+
+type Request = Vec<u8>;
+type Response = Vec<u8>;
+
+struct Message {
+    // Arbitrary message data.
+    data: Vec<u8>,
```

Comment:
This data is always the request, right? If so, then perhaps just call it `request`, also for symmetry with the field below.